tools/summarise_analyse.py

```diff
@@ -0,0 +1,161 @@
+#! /usr/bin/env python3
+
+"""Produce simple statistics on the results of 'charmcraft analyse'."""
+
+import collections
+import logging
+import operator
+import pathlib
+import re
+import subprocess
+import typing
+
+import click
+import rich.console
+import rich.logging
+from helpers import count_and_percentage_table
+from helpers import iter_repositories
+
+logger = logging.getLogger(__name__)
+
+
+def analyse_repo(repo: pathlib.Path, repack: bool):
+    """Run 'charmcraft analyse' on the provided repository."""
+    logger.info(f"Analysing {repo}")
+    results = collections.defaultdict(set)
+    # Make sure that the charm is packed.
+    if repack or not tuple(repo.glob("*.charm")):
+        try:
+            subprocess.run(
+                ["charmcraft", "pack"], check=True, cwd=repo, capture_output=True
+            )
+        except subprocess.CalledProcessError as e:
+            # It might be interesting to look into these, but it seems out of scope for analyse.
+            logger.error(
+                "Failed to pack charm: %s: %s (%r/%r)", repo, e, e.stdout, e.stderr
+            )
+            return results
+        # Remove the build environment - otherwise, this ends up using a huge amount
+        # of disk space (in /var/snap/lxd).
+        subprocess.run(["charmcraft", "clean"], check=True, cwd=repo)
+    continuation = None
+    for charm in repo.glob("*.charm"):
```

Comment:
If you run with the `repack` option, and there are already `.charm` files with different names, this will get both, right? But I guess the expectation is that the only charm file is the one you made?

It doesn't seem like you can set the output filename, but you could use `charmcraft pack`'s `--output` flag to set an output directory and remove the directory before repacking ... `./.cache/$charm/.packed/` maybe? or maybe a parallel structure like `./.packed-cache/$charm/` would be more convenient since you can easily remove all the packed charms in one operation

Reply:
Yeah, I'm somewhat assuming that the cache folder is only used by these tools, or is cleaned up to be that way. And yes, I really wanted to just get the output name but it doesn't seem possible, probably because of the history of charmcraft being able to produce multiple charms in a pack.

Putting them all somewhere else is an interesting idea, though.